lib/api/core.dart

```diff
@@ -192,10 +193,20 @@ class ApiConnection {
   }
 
   Future<T> postFileFromStream<T>(String routeName, T Function(Map<String, dynamic>) fromJson,
-      String path, Stream<List<int>> content, int length, {String? filename}) async {
+      String path, Stream<List<int>> content, int length, {String? filename,
+      String? contentType}) async {
     final url = realmUrl.replace(path: "/api/v1/$path");
+    MediaType? effectiveContentType;
+    if (contentType != null) {
+      try {
+        effectiveContentType = MediaType.parse(contentType);
```

Comment:
I think of the "effectiveFoo" naming seen in Flutter code as referring to a fallback cascade: e.g. it's the value passed on this widget, or else its default from the ambient theme, or else the value from the default theme.

This is a different sort of thing: it's the value given, but parsed. A good name could be `parsedContentType`.